packages/@aws-cdk/aws-stepfunctions-tasks/test/sagemaker/create-training-job.test.ts

```diff
@@ -179,7 +179,7 @@ test('create complex training job', () => {
     },
     resourceConfig: {
       instanceCount: 1,
-      instanceType: ec2.InstanceType.of(ec2.InstanceClass.P3, ec2.InstanceSize.XLARGE2),
+      instanceType: new ec2.InstanceType(sfn.JsonPath.stringAt('$.TrainingJob.InstanceType')),
       volumeSize: cdk.Size.gibibytes(50),
       volumeEncryptionKey: kmsKey,
```

Comment:
please add a new test for exercising the added control flow rather than to modify an existing test. 
we want this one to stay as it is and ensure that it passes doesn't regress.

Reply:
I'm happy to add a separate test case if required, but I'm not sure it ads any coverage.  Here are the reasons why I went this route instead:

- Stays consistent with [CreateSageMakerEndpointConfig](https://github.com/aws/aws-cdk/blob/0556e6b710c53dacf6b65926b4cc5b82fb082ee6/packages/%40aws-cdk/aws-stepfunctions-tasks/test/sagemaker/create-endpoint-config.test.ts#L72) (added in [this similar commit]( https://github.com/aws/aws-cdk/commit/5fb0ea6276db26f76b99b1826d742eae979f4ed9))
- Test case for the path where InstanceType is provided at build time is already covered [here](https://github.com/aws/aws-cdk/blob/0556e6b710c53dacf6b65926b4cc5b82fb082ee6/packages/%40aws-cdk/aws-stepfunctions-tasks/test/sagemaker/create-training-job.test.ts#L323).

Reply:
It does add coverage because there are 2 scenarios that we want to test here wrt the generated Amazon States Language (ASL) definition:
1. providing a literal string as an instance type - the `key` will be `InstanceType` in the resource config
2. providing a dynamic reference as an instance type - the `key` will be `InstanceType.$` in the resource config

The prior art here is less reusable because the classes that were initially introduced had an `@experimental` annotation, but since those were dropped recently, we want to take some additional measures to ensure we're not regressing.

by switching the test over from literal to dynamic, we're still only capturing 1 out of the 2 scenarios so ideally we have tests that cover both. Let me know if you have any questions.

Reply:
discussed offline - both of these scenarios are covered in tests as other existing tests already cover the literal string use case.